aten/src/ATen/native/AutogradComposite.cpp

```diff
@@ -91,14 +93,25 @@ bool _has_same_storage_numel(const at::Tensor& base, const at::Tensor& other) {
   return base.storage().sym_nbytes() / base.itemsize() == other.storage().sym_nbytes() / other.itemsize();
 }
 
-Tensor _lazy_clone(Tensor const& self) {
+Tensor _lazy_clone(Tensor const& self, optional<c10::Device> device_opt) {
+  optional<c10::Allocator*> allocator_opt = nullopt;
+  if (device_opt.has_value()) {
+    allocator_opt = at::empty({}, at::TensorOptions().device(device_opt.value())).storage().allocator();
```

Comment:
This is not an ideal way to get the allocator, but there currently is no easy way to do it.